mindmeld/core.py

```diff
@@ -529,9 +531,15 @@ def _get_form_details(query_span, offset, form_in, form_out):
             span_out = query.transform_span(query_span, form_in, form_out)
             full_text = query.get_text_form(form_out)
             text = span_out.slice(full_text)
-            tok_start = len(full_text[: span_out.start].split())
+            tok_start = 0
+            # The span range is till the span_out or max to the second last char
+            span_range = min(span_out.start, len(full_text) - 1)
+            for idx, current_char in enumerate(full_text[:span_range]):
+                # Increment the counter only if a whitespace follows a non-whitespace
+                next_char = full_text[idx + 1]
+                if current_char != WHITESPACE_CHAR and next_char == WHITESPACE_CHAR:
```

Comment:
Consider using `string.isspace()` from the `string` library. This would check for tabs, spaces, newline etc and returns True if there are only such whitespace characters in the string. With this, you may not need to store the 'WHITESPACE_CHAR' as a constant. [Source](https://www.programiz.com/python-programming/methods/string/isspace)

Reply:
Great suggestion Kunal. I am still not sure how this would affect non-empty space tokenized languages though.

Reply:
I implemented a way of accounting for token spans based on the normalized tokens, that should be resilient to non-space tokenizers. Thoughts?

Reply:
Good point. Using the normalized tokens is definitely more robust here as compared to a whitespace delimiter. This should work with the upcoming changes of `character tokenization` and `spacy tokenization` as each individual token can be found in `query._normalized_tokens` as you have in your PR

Reply:
So after much debugging, I had to revert the normalized token changes. The reason for this is as follows:
1. This function operates on raw, processed and normalized text. So the token index from the raw text is different to that of a normalized text. For example, the text `meet me from 9am...10am` is normalized as `meet me from 9am 10am`, the token index of `10am` in the raw text is 3 and the token index of the normalized text is 4.
2. The split function used here is done here since once we tokenize, we join using a whitespace to represent the normalized text, so we do this `meet me from 9am...10am` -> `[meet, me, from, 9am, 10am]` -> `meet me from 9am 10am`. So any non-empty space tokenization operation will add whitespace by the time this function operates on it.

For the above two reasons, I have reverted to operating on empty space tokens.